### PR TITLE
LEDE-2650 Newsletter Button Block

### PR DESCRIPTION
**Pull request by cahdeemer**
## Summary
Revises supports for core button blocks and add custom Newsletter Button block that provides an email-friendly wrapper for core button block.

## Notes for reviewers
* Uses similar methods / pattern to Newsletter Heading, Newsletter List, and Newsletter Paragraph blocks.
* Version bump

## Ticket
https://alleyinteractive.atlassian.net/browse/LEDE-2650

### CODE REVIEW

File: hooks/useInnerBlocks/index.js
Comment (efuller) [inline hooks/useInnerBlocks/index.js:1]: Nice!

File: plugin.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * Plugin Name: Newsletter Builder
  * Plugin URI: https://github.com/alleyinteractive/wp-newsletter-builder
  * Description: Interface to manage email newsletters
- * Version: 0.3.22
+ * Version: 0.3.23
```
Comment (efuller) [inline plugin.php:6]: I forgot to mention earlier that we now have an npm script that will create releases! You can just run `npm run release` and it'll walk you through bumping the version!

File: blocks/button/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    margin: '0 auto',
+    width: btnWidth,
+  };
+
+  useEffect(() => {
+    setAttributes({
+      radius: innerBorderRadius,
+      btnWidth: innerWidth !== undefined ? `${innerWidth}%` : 'max-content',
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline blocks/button/edit.tsx:62]: Is disabling this ESLint rule still needed? It looks like all dependencies are listed. Could we remove?

**Review by efuller — APPROVED** [review]
⚡ Couple of comments/questions, but nothing blocking! Buttons ftw

File: blocks/button/edit.tsx (hunk 2 above)
Comment (cahdeemer) [reply blocks/button/edit.tsx:62]: eslint wants `setAttributes` to be a part of the dependency array, so we still need this unfortunately

File: plugin.php (hunk 1 above)
Comment (cahdeemer) [reply plugin.php:6]: Awesome! Thanks!

**Pull request merged**